core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -146,9 +145,12 @@ public T acquire(Integer time, TimeUnit unit) {
    * @param resource the resource to be released, it should not be reused after calling this method
    */
   public void release(T resource) {
-    mResources.add(resource);
-    try (LockResource r = new LockResource(mTakeLock)) {
-      mNotEmpty.signal();
+    if (resource != null) {
+      mResources.add(resource);
+      mCurrentCapacity.incrementAndGet();
```

Comment:
`mCurrentCapacity` is the total number of objects in the pool, whether acquired or still available. A method should never decrease it.